### DISCUSSION TITLE
How can I modify rg (ripgrep) to perform case sensitive search only when the given PATTERN has any uppercase char?

### DISCUSSION THREAD

**Discussion by tifDev** [BurntSushi/ripgrep#1594] [General] [answered]
Some comands like `fd` automatically detects if the PATTERN has any uppercase letter it performs a case sensitive search, otherwise it performs a case insentive search. I want to apply this feature to `rg` so that the default is case insensitive and only if the given PATTERN has any Uppercase then it performs a case sensitive search.

**How can I achieve this?**

I tried to add this code as an alias but it didn't worked as expected:
```

rg () {
  if [[ "$1" =~ [A-Z] ]] || [[ "$2" =~ [A-Z] ]] || [[ "$3" =~ [A-Z] ]]; then
    # has uppercase
    rg
  else
    # no uppercase
    rg -i
  fi
}

```

**Comment by BurntSushi** [comment, marked as answer]
You don't need to implement this yourself. ripgrep already has a flag for this: `-s/--smart-case`:

```
Searches case insensitively if the pattern is all lowercase.
Search case sensitively otherwise.

This overrides the -s/--case-sensitive and -i/--ignore-case flags.
```

If you want this enabled by default, then you can either use a function (like you were trying), an alias (i.e., `alias rg="rg -S"`) or create a [configuration file](https://github.com/BurntSushi/ripgrep/blob/a700b758436548560aab16609e998c32180c824a/GUIDE.md#configuration-file) and put it there.

Also, in the future, don't just say something "didn't work as expected." Please be sure to provide an example input along with its output _and_ the output you expected. Otherwise, people have to guess what it is that you meant by "expected."
